requests/sessions.py

```diff
@@ -251,13 +251,25 @@ def resolve_redirects(self, resp, req, stream=False, timeout=None,
                 url = self.get_redirect_target(resp)
                 yield resp
 
-    def rebuild_auth(self, prepared_request, response):
+    def rebuild_auth(self, prepared_request, response, proxies):
```

Comment:
This is a functionally public API. We can't change the signature like this

Reply:
Oh my bad, I thought this was internal. I'll look into for an alternative approach

Reply:
I can just extract this change to a different (private?) function and call it before `rebuild_auth`. WDYT?